From f34e1909d879c585bfe85d964dfb518238058f19 Mon Sep 17 00:00:00 2001
From: Vince Chang <vince_chang@aspeedtech.com>
Date: Tue, 25 Mar 2025 14:11:02 +0800
Subject: [PATCH 2/2] Fix build issue updatde to C++23

Fix below build fail.
```
recipe-sysroot/usr/include/sdbusplus/asio/connection.hpp:88:32: error: 'move_only_function' in namespace 'std' does not name
a template type
|    88 |     using send_function = std::move_only_function<callback_t>;
```

Upstream-Status: Submitted [https://gerrit.openbmc.org/c/openbmc/pfr-manager/+/79154]

Change-Id: Ib88e44e65facce5bb4497fdc82f07c94900e16cd
Signed-off-by: Vince Chang <vince_chang@aspeedtech.com>
---
 CMakeLists.txt         | 2 +-
 libpfr/CMakeLists.txt  | 2 +-
 service/CMakeLists.txt | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 913860b..9766a14 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
 cmake_minimum_required(VERSION 2.8.10 FATAL_ERROR)
-set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD 23)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/libpfr/inc)
diff --git a/libpfr/CMakeLists.txt b/libpfr/CMakeLists.txt
index dfa0ad2..c5089bd 100644
--- a/libpfr/CMakeLists.txt
+++ b/libpfr/CMakeLists.txt
@@ -1,6 +1,6 @@
 cmake_minimum_required(VERSION 2.8.10 FATAL_ERROR)
 project(pfr CXX)
-set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD 23)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 include(ExternalProject)
diff --git a/service/CMakeLists.txt b/service/CMakeLists.txt
index ce08540..d42ca13 100644
--- a/service/CMakeLists.txt
+++ b/service/CMakeLists.txt
@@ -1,6 +1,6 @@
 cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
 project(pfr-manager CXX)
-set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD 23)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 # Temporarily disable rtti
 # See https://github.com/chriskohlhoff/asio/issues/533
-- 
2.43.0

